src/signatures.rs

```diff
@@ -22,8 +22,48 @@ use rkyv::{Archive, Deserialize, Serialize};
 pub struct Signature(pub(crate) G1Affine);
 
 impl Signature {
-    /// Aggregate a set of signatures by simply adding up the points.
-    pub fn aggregate(&self, sigs: &[Signature]) -> Self {
+    /// Returns true if the inner point is valid according to certain criteria.
+    ///
+    /// A [`Signature`] is considered valid if its inner point meets the
+    /// following conditions:
+    /// 1. It is free of an $h$-torsion component and exists within the
+    ///    $q$-order subgroup $\mathbb{G}_2$.
+    /// 2. It is on the curve.
+    /// 3. It is not the identity.
+    pub fn is_valid(&self) -> bool {
+        is_valid(&self.0)
+    }
+}
+
+impl Serializable<48> for Signature {
+    type Error = Error;
+
+    fn to_bytes(&self) -> [u8; Self::SIZE] {
+        self.0.to_bytes()
+    }
+
+    fn from_bytes(bytes: &[u8; Self::SIZE]) -> Result<Self, Error> {
+        Ok(Self(G1Affine::from_bytes(bytes)?))
+    }
+}
+
+pub(crate) fn is_valid(sig: &G1Affine) -> bool {
+    let is_identity: bool = sig.is_identity().into();
+    sig.is_torsion_free().into() && sig.is_on_curve().into() && !is_identity
+}
+
+/// A BLS signature, in the multi-signature scheme.
+#[derive(Default, Copy, Clone, Debug, Eq, PartialEq)]
+#[cfg_attr(
+    feature = "rkyv-impl",
+    derive(Archive, Deserialize, Serialize),
+    archive_attr(derive(bytecheck::CheckBytes))
+)]
+pub struct MultisigSignature(pub(crate) G1Affine);
```

Comment:
Doesn't feel redundant to call it Multi signature signature?

Reply:
I feel like I would then also have to change the name of `MultisigPublicKey` to have the pair of:

- `MultiPublicKey`
- `MultiSignature` 

I don't like it though, since the concept of "multi-sig" refers specifically to the scheme, not the structure, and erasing it would kind of defeat the purpose of this PR.